lib/src/runner/browser/browser.dart

```diff
@@ -126,6 +126,8 @@ abstract class Browser {
     });
 
     // Swallow exceptions. The user should explicitly use [onExit] for these.
-    return onExit.catchError((_) {});
+    // However, surface application exceptions which are thrown due to startup
+    // issues with the browser.
+    return onExit.catchError((_) {}, test: (e) => e is! ApplicationException);
```

Comment:
Indeed. What I intended to say is that the issue raised in this pull request is indeed an actual issue which you'll be looking at. I will close this pull until I can get a better handle on what is causing the flakiness.